ext/openssl/openssl.c

```diff
@@ -703,7 +706,14 @@ static void add_assoc_name_entry(zval * val, char * key, X509_NAME * name, int s
 				add_assoc_stringl(&subitem, sname, (char *)to_add, to_add_len);
 			}
 		}
+
+		if (needs_free) {
+			/* ASN1_STRING_to_UTF8(3): The buffer out should be freed using free(3) */
+			free(to_add);
+			needs_free = 0;
```

Comment:
I don't understand why.

The same `char *` is assigned to by two different processes, one that allocs a new buffer, and one that doesn't.

If I remove the condition we get a double free when it takes the path that doesn't alloc a new buffer.